File: ruby/examples/openai_chat.rb
# frozen_string_literal: true

require 'buble'

client = Buble::Client.new

completion = client.chat.completions.create(
  model: 'openai/gpt-5.4',
  messages: [
    { role: 'user', content: 'Write a short launch summary.' }
  ],
  max_completion_tokens: 800
)

puts completion.dig('choices', 0, 'message', 'content')

File: rust/examples/openai_chat.rs
use buble::{Client, Result};
use serde_json::json;

#[tokio::main]
async fn main() -> Result<()> {
    let client = Client::from_env()?;

    let completion = client
        .chat()
        .completions()
        .create(json!({
            "model": "openai/gpt-5.4",
            "messages": [
                { "role": "user", "content": "Write a short launch summary." }
            ],
            "max_completion_tokens": 800
        }))
        .await?;

    println!("{completion:#?}");
    Ok(())
}
